### ISSUE TITLE
Emphasize keyword over fielddata as a solution to aggregating or sorting on String fields

### ISSUE THREAD

**Issue by sachinfrayne** [elastic/elasticsearch#46695]
Change the order of the fielddata error message to emphasize keyword as the preferred solution rather than as an alternative for the majority of cases where a user has mapped the field to the `text` dataType but had the intention of using that field instead for aggregations and/or sorting.

```
{
"error": {
    "root_cause": [
      {
        "type": "illegal_argument_exception",
"reason": "Fielddata is disabled on text fields by default. Set fielddata=true on [geoip.country_name] in order to load fielddata in memory by uninverting the inverted index. Note that this can however use significant memory. Alternatively use a keyword field instead."
} ],
...
```

**Comment by jtibshirani** [comment]
@sachinfrayne this is a nice suggestion, would you be up for submitting a PR to change the error message?

**Comment by sachinfrayne** [comment]
Hi @jtibshirani 

Yes, I will write it up this evening and submit it, :).

Which branch would you recommend I submit it to?

**Comment by jtibshirani** [comment]
Great -- it would be best to open a PR against the master branch, then I can make sure it gets backported to the right release branches.

**Comment by sachinfrayne** [comment]
@jtibshirani, I just recalled that I wanted to make this change, I found the ticket again and I thought I had submitted a comment but I must have forgotten to press the comments button.

Can you please have a look at this message and let me know what you think:

> Text fields are not optimized for aggregations and/or sorting and therefore these operations are disabled by default. Please use a field with a keyword datatype instead. Otherwise Set fielddata=true on [FIELD] in order to load fielddata in memory by uninverting the inverted index. Note that this can however use significant memory.

**Comment by jtibshirani** [comment]
@sachinfrayne that looks like a nice start to me, feel free to open a PR and I can give any detailed comments there.

**Comment by jtibshirani** [comment]
Closing out, this was addressed in #52753.
